### PR TITLE
Release 8.2.0

### PR DESCRIPTION
**Pull request by camundait**
Merge the release branch with main. This pull request was created by Zeebe :rocket:.

### CODE REVIEW

**Comment by npepinpe** [comment]
haha, this makes me the committer when I fix conflicts? That's one way to get around the CLA ;)

**Comment by npepinpe** [comment]
Known flaky test around the local stack integration tests.

**Comment by npepinpe** [comment]
Seems unnecessary that all the tests are running twice, for push and pull_request :shrug:

**Comment by npepinpe** [comment]
It's still running everything twice...?

File: .github/workflows/dispatch-release-8-2.yaml
Comment (megglos) [inline .github/workflows/dispatch-release-8-2.yaml:13]: 🔧 I think we also need to add a new workflow `dispatch-release-8-3.yaml` for the new 8.3 minor going forward.

Maybe we need to rephrase this [form](https://github.com/zeebe-io/zeebe-engineering-processes/blame/master/src/main/resources/release/forms/post_release_update_dry_stable_release_workflow.form#L20) a bit, highlighting the current released one needs and update  a new one for the new development version needs to created 🤔

File: .github/workflows/release-stable-dry-run.yml
Comment (megglos) [inline .github/workflows/release-stable-dry-run.yml:17]: ❌ You need a distinct version for each job, otherwise the artifact upload might encounter conflicts (I remember I had those during testing)
```suggestion
      releaseVersion: 0.8.2
```

**Review by megglos — APPROVED** [review]
🎉  Just two things I noticed to take a look at before merging, pre-approved!

File: .github/workflows/dispatch-release-8-2.yaml
Comment (npepinpe) [reply .github/workflows/dispatch-release-8-2.yaml:13]: Ah, I wasn't aware of it. I was wondering why the workflow already existed to be honest :sweat_smile:

Comment (npepinpe) [reply .github/workflows/dispatch-release-8-2.yaml:13]: I'll update the release process as well :+1:

**Comment by npepinpe** [comment]
bors merge

**Pull request merged**
